syntax = "proto3";

package lint;

service LintProto {
  rpc SendLint (LintRequest) returns (LintReply) {}
}

message LintRequest {
  string name = 1;
  repeated LintFile lintFiles = 2;
  LintMeta lintMeta = 3;
  LintPatch lintPatch = 4;
}

message LintFile {
  string path = 1;
  bytes content = 2;
}

message LintMeta {
  string path = 1;
  bytes content = 2;
}

message LintPatch {
  string path = 1;
  bytes content = 2;
}

message LintReply {
  string name = 1;
  repeated LintReport lintReports = 2;
}

message LintReport {
  string file = 1;
  int64 line = 2;
  string type = 3;
  string details = 4;
}
